### PR TITLE
feat: remember window position and size of Podman Desktop window

### LINKED ISSUE #6115: Remember window state and position from last start
### Is your enhancement related to a problem? Please describe

Reported via feedback form: every time you start the main window reverts to being open and in the default position.

### Describe the solution you'd like

When you exit Podman Desktop we should save the last window state (position/size, or closed) and reopen in the same state.

### Describe alternatives you've considered

_No response_

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by benoitf**
### What does this PR do?
Save and restore the window position where it was before closing Podman Desktop

### What issues does this PR fix or reference?

fixes https://github.com/containers/podman-desktop/issues/6115

### How to test this PR?

start podman desktop, move and resize the window. Then exit and restart
Check that it's at the same position than before

you can disable it through the settings/preferences (window)

- [x] Tests are covering the bug fix or the new feature

### CODE REVIEW

**Review by deboer-tim — COMMENTED** [review]
Tested & worked, with multiple monitors.

A couple times the window appeared in the default position and then jumped to the correct position. Is there anything we can do to avoid this/set position earlier?

I don't love creating a new Window preferences category as it feels like we have a lot of single preferences in their own category. There isn't anything obvious to group this with though either. Maybe we should start a General category?

Would also be nice if we remembered if the window was closed on last run (i.e. system tray only), but ok if that's a separate issue. (I think we have one in the backlog, and maybe we should still show the loading screen or something so that you know it actually launched)

File: packages/main/src/system/window/window-init.ts [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+  constructor(configurationRegistry: IConfigurationRegistry) {
+    this.#configurationRegistry = configurationRegistry;
+  }
+
+  init(): void {
+    const heightKey = `${WindowSettings.SectionName}.${WindowSettings.Height}`;
+    const widthKey = `${WindowSettings.SectionName}.${WindowSettings.Width}`;
+    const xPositionKey = `${WindowSettings.SectionName}.${WindowSettings.XPosition}`;
+    const yPositionKey = `${WindowSettings.SectionName}.${WindowSettings.YPosition}`;
+    const restorePosition = `${WindowSettings.SectionName}.${WindowSettings.RestorePosition}`;
```
Comment (jeffmaury) [inline packages/main/src/system/window/window-init.ts:37]: Seems to me it should end by Key to be aligned with previous lines

**Comment by cdrage** [comment]
Tested and works with macOS, but will need to test on Linux and Windows too.

On M1, it's hard to take a video of it, but I get the same thing where it "jumps" to the separate section of the screen.

EDIT: Also need to check plugging in two monitors, putting PD on one monitor, unplugging it and then seeing what happens with PD. Usually some windows will go "back" to the main screen after.

**Review by jeffmaury — COMMENTED** [review]
I think the split between window init and window handler are making the code less readable and the first one creates conf that is used by the second so a merge would be good.

I have also a concern with storing the position as I have the following use case:
- Working with 2 screens Podman Desktop starts on screen 1 (laptop) and moved to screen 2 (monitor)
- Going back home (laptop only) restarting Podman Desktop but no window being visible

**Pull request closed without merging**